### ISSUE TITLE
until not templating variables in every loop execution

### ISSUE THREAD

**Issue by cathalkragan** [ansible/ansible#79171]
### Summary

When retrying a task using until I was expecting the templated variables to be re-generated with each loop execution, but that's not the case.  Ansible looks be templating all the variables just once and using them with all the retries.

### Issue Type

Bug Report

### Component Name

ansible-core

### Ansible Version

```console
$ ansible --version
ansible [core 2.13.4]
  config file = None
  configured module search path = ['/Users/dev/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dev/Library/Python/3.8/lib/python/site-packages/ansible
  ansible collection location = /Users/dev/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/dev/Library/Python/3.8/bin/ansible
  python version = 3.8.2 (default, Dec 21 2020, 15:06:04) [Clang 12.0.0 (clang-1200.0.32.29)]
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
ANSIBLE_FORCE_COLOR(env: ANSIBLE_FORCE_COLOR) = False
```

### OS / Environment

macOS

### Steps to Reproduce

```yaml (paste below)
---
- name: Until
  hosts: localhost
  gather_facts: false

  tasks:
    - name: Print time
      ansible.builtin.debug:
        msg: "{{ lookup('pipe', 'date +\"%Y-%m-%d %H:%M:%S\"') }}"
      until: false
      retries: 2
      delay: 10
```

### Expected Results

I was expecting to see the time stamps change with each execution of the loop, especially since I'm using a lookup to get the date from the system.  But looks like the msg variable is being templated only once.

### Actual Results

```console
ansible-playbook [core 2.13.4]
  config file = None
  configured module search path = ['/Users/dev/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dev/Library/Python/3.8/lib/python/site-packages/ansible
  ansible collection location = /Users/dev/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/dev/Library/Python/3.8/bin/ansible-playbook
  python version = 3.8.2 (default, Dec 21 2020, 15:06:04) [Clang 12.0.0 (clang-1200.0.32.29)]
  jinja version = 3.1.2
  libyaml = True
No config file found; using defaults
setting up inventory plugins
host_list declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
script declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
yaml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
ini declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
Skipping due to inventory source not existing or not being readable by the current user
toml declined parsing /etc/ansible/hosts as it did not pass its verify_file() method
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
Loading callback plugin default of type stdout, v2.0 from /Users/dev/Library/Python/3.8/lib/python/site-packages/ansible/plugins/callback/default.py
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.

PLAYBOOK: until.yml ***********************************************************************************************************
Positional arguments: Ansible/until.yml
verbosity: 4
connection: smart
timeout: 10
become_method: sudo
tags: ('all',)
inventory: ('/etc/ansible/hosts',)
forks: 5
1 plays in Ansible/until.yml

PLAY [Until] ******************************************************************************************************************
META: ran handlers

TASK [Print time] *************************************************************************************************************
task path: /Users/dev/Library/Mobile Documents/com~apple~CloudDocs/Programming/Ansible/until.yml:7
FAILED - RETRYING: [localhost]: Print time (2 retries left).Result was: {
    "attempts": 1,
    "changed": false,
    "msg": "2022-10-20 07:47:29",
    "retries": 3
}
FAILED - RETRYING: [localhost]: Print time (1 retries left).Result was: {
    "attempts": 2,
    "changed": false,
    "msg": "2022-10-20 07:47:29",
    "retries": 3
}
fatal: [localhost]: FAILED! => {
    "msg": "2022-10-20 07:47:29"
}

PLAY RECAP ********************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

### Code of Conduct

- [x] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
None

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/issues/78266
